cmd/helm/push.go

```diff
@@ -40,6 +40,8 @@ type registryPushOptions struct {
 	caFile                string
 	insecureSkipTLSverify bool
 	plainHTTP             bool
+	password              string
+	username              string
```

Comment:
Too late now, but I thought the idea was to authenticate before using push

Reply:
> authenticate before using push

Do you mean `helm registry login` then `helm push` or are you referring to a logic in code?

Reply:
Yeh, I thought these commands were set up to authenticate with `helm login` and all the other commands need to deal with TLS etc, but not authn Sounds like my assumption was incorrect.

Reply:
> Yeh, I thought these commands were set up to authenticate with `helm login` and all the other commands need to deal with TLS etc, but not authn Sounds like my assumption was incorrect.

You can do both. You can login to allow multiple invocations without having the explicitly specifying each option, or one can integrate ad-hoc, as enabled through this enhancement